File: Semester-1/Pemrograman-1/Modul-6/C/PRAK605-2310817210014-DamarjatiSuryoLaksono.c
#include <stdio.h>

void perkalianMatriks(int matriksPertama[][10], int matriksKedua[][10], int hasil[][10], int n) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            hasil[i][j] = 0;
            for (int k = 0; k < n; k++) {
                hasil[i][j] += matriksPertama[i][k] * matriksKedua[k][j];
            }
        }
    }
}

void tampilkanMatriks(int matriks[][10], int n) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            printf("%d ", matriks[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int n, matriksPertama[10][10], matriksKedua[10][10], hasil[10][10];
    scanf("%d", &n);
    printf("Matriks A\n");
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            scanf("%d", &matriksPertama[i][j]);
        }
    }
    printf("Matriks B\n");
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            scanf("%d", &matriksKedua[i][j]);
        }
    }
    perkalianMatriks(matriksPertama, matriksKedua, hasil, n);
    printf("Matriks AXB\n");
    tampilkanMatriks(hasil, n);

    return 0;
}

File: Semester-1/Pemrograman-1/Modul-6/Python/PRAK605-2310817210014-DamarjatiSuryoLaksono.py
def perkalian_matriks(matriks_a, matriks_b):
  if len(matriks_a[0]) != len(matriks_b):
    raise ValueError("Dimensi matriks tidak sesuai")
  matriks_hasil = [[0 for _ in range(len(matriks_b[0]))]
                  for _ in range(len(matriks_a))]
  for i in range(len(matriks_hasil)):
    for j in range(len(matriks_hasil[0])):
      for k in range(len(matriks_a[0])):
        matriks_hasil[i][j] += matriks_a[i][k] * matriks_b[k][j]

  return matriks_hasil
baris = int(input(""))

print("Matriks A")
matriks_a = []
for i in range(baris):
  matriks_a.append(list(map(int, input().split())))
print("Matriks B")
matriks_b = []
for i in range(baris):
  matriks_b.append(list(map(int, input().split())))
matriks_hasil = perkalian_matriks(matriks_a, matriks_b)
print("Matriks AXB")
for i in range(baris):
  for j in range(len(matriks_hasil[0])):
    print(matriks_hasil[i][j], end=" ")
  print()
